integration-tests/test_satellite.py

```diff
@@ -0,0 +1,113 @@
+        assert archive_result.returncode == 0
+
+        # extract the collected tar file
+        archive_location = archive_result.stdout.split("\n")[-2].split(" ")[-1]
+        cmd = ["tar", "-xzvf", archive_location]
+        subprocess.check_call(cmd)
+
+        # the redaction data should not locate in the collected files
+        archive_extract = archive_location.split(".")[0].split("/")[-1]
+        cmd = "cd %s; grep -rnw . -e %s" % (archive_extract, hostname_to_test)
+        # should not find the obfuscate pattern, the return code should not zero.
+        returncode = subprocess.check_call(cmd, shell=True)
+        assert returncode
+    except Exception as e:
+        traceback.print_exc()
+    finally:
+        # Recover the previous hostname for the system
+        set_hostname(previous_hostname)
+        # Remove all data redaction
+        cmd = ["rm", "-f", "/etc/insights-client/file-content-redaction.yaml"]
+        subprocess.check_call(cmd)
+
+
+def test_set_ansible_host_info(insights_client):
+    """
+    Test if the ansible-host can be set with satellite
+    Related to https://issues.redhat.com/browse/RHELPLAN-160348,
+    and https://bugzilla.redhat.com/show_bug.cgi?id=222729
+    """
+    # Register system against Satellite, and register insights through satellite
+    insights_client.register()
+    sleep(10)
+    assert insights_client.is_registered
+
+    # Update ansible-host
+    set_ansible_host = insights_client.run("--ansible-host='foo.example.com'")
+    assert set_ansible_host.returncode == 0
+
+
+def test_register_group_option(insights_client):
+    """
+    Test registering host when using --group option through satellite
+    Related to https://bugzilla.redhat.com/show_bug.cgi?id=2059031,
+    and https://issues.redhat.com/browse/RHEL-3739
+    """
+    # make sure the system is not registered to insights
+    with contextlib.suppress(Exception):
+        insights_client.unregister()
+    assert not insights_client.is_registered
+    """ Register the system to insights with group option
+    Use insights_client.run can not find the existing bug, so try subprocess
+    register_with_group = insights_client.run("--register", "--group='tag'")
+    assert register_with_group.returncode == 0
+    """
+    cmd = ["insights-client", "--register", "--group='tag'"]
+    subprocess.check_call(cmd, shell=True)
+
+
+def set_hostname(hostname_to_set):
+    cmd = ["hostnamectl", "set-hostname", hostname_to_set]
+    subprocess.check_call(cmd)
+
+
+def data_redaction(pattern_string):
+    insights_folder = "/etc/insights-client/"
+    if not os.path.exists(insights_folder):
+        os.makedirs(insights_folder)
+    file_content = {"patterns": {"regex": [pattern_string]}}
+    data_redaction_file = "/etc/insights-client/file-content-redaction.yaml"
+    with open(data_redaction_file, "w") as f:
+        yaml.dump(file_content, f, default_flow_style=False)
+    os.chmod(data_redaction_file, int("600", 8))
```

Comment:
while `int("600", 8)` works, IMHO it is not that readable :-) simply use the Python prefixes for non-decimal numbers:
```suggestion
    os.chmod(data_redaction_file, 0o600)
```